### PR TITLE
Avoid return 503 after closing fastify

### PR DESCRIPTION
**Pull request by allevo**
Address #1714 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by allevo** [comment]
Tests fixed

File: lib/route.js [hunk 1]
```diff
@@ -267,20 +267,9 @@ function buildRouting (options) {
...
-        'Content-Length': '80'
-      }
-      if (req.httpVersionMajor !== 2) {
-        headers.Connection = 'close'
-      }
-      res.writeHead(503, headers)
-      res.end('{"error":"Service Unavailable","message":"Service Unavailable","statusCode":503}')
       if (req.httpVersionMajor !== 2) {
-        // This is not needed in HTTP/2
-        setImmediate(() => req.destroy())
```
Comment (delvedor) [inline lib/route.js]: Would you mind restoring this but putting it in `reply.res.on('finish')`?

**Comment by allevo** [comment]
Oki, I'll add the flag

File: lib/route.js (hunk 1 above)
Comment (allevo) [reply lib/route.js]: done

**Comment by allevo** [comment]
Flag added: `return503OnClosing`

File: lib/route.js [hunk 2]
```diff
@@ -267,20 +269,27 @@ function buildRouting (options) {
...
+          'Content-Length': '80'
+        }
+        if (req.httpVersionMajor !== 2) {
+          headers.Connection = 'close'
+        }
+        res.writeHead(503, headers)
+        res.end('{"error":"Service Unavailable","message":"Service Unavailable","statusCode":503}')
+        if (req.httpVersionMajor !== 2) {
+          // This is not needed in HTTP/2
+          setImmediate(() => req.destroy())
```
Comment (delvedor) [inline lib/route.js]: I think we can use the res event also here.
```suggestion
          res.once('finish', () => req.destroy())
```

Comment (allevo) [reply lib/route.js]: addressed with some rework

File: test/http2/closing.js [hunk 3]
```diff
@@ -6,43 +6,97 @@ const Fastify = require('../..')
...
+  }
+
+  fastify.get('/', () => Promise.resolve({}))
+
+  fastify.listen(0, err => {
+    t.error(err)
+    fastify.server.unref()
+
+    // Skipped because there is likely a bug on Node 8.
+    test('return 200', { skip: semver.lt(process.versions.node, '10.15.0') }, t => {
```
Comment (Eomm) [inline test/http2/closing.js]:
```suggestion
    t.test('return 200', { skip: semver.lt(process.versions.node, '10.15.0') }, t => {
```

File: test/http2/closing.js [hunk 4]
```diff
@@ -6,43 +6,97 @@ const Fastify = require('../..')
...
```
Comment (Eomm) [inline test/http2/closing.js]:
```suggestion
    t.test('return 200', { skip: semver.lt(process.versions.node, '10.15.0') }, t => {
```

**Pull request merged**